protocol/epoch_test.go

```diff
@@ -0,0 +1,280 @@
+				require.True(t, errors.Is(err, ErrInvalidEpochThrottled))
+			},
+		},
+		{
+			"aborts sync if the epoch is backdated",
+			func(t *testing.T, setup *epochTestSetup) {
+				cfg := &UpdateConfig{
+					Mux:       setup.tp.LocalMux,
+					DB:        setup.ls.DB,
+					BlobStore: setup.ls.BlobStore,
+					Item: &UpdateQueueItem{
+						PeerIDs: NewPeerSet([]crypto.Hash{
+							crypto.HashPub(setup.tp.RemoteSigner.Pub()),
+						}),
+						Name:        name,
+						EpochHeight: 0,
+					},
+				}
+				require.NoError(t, store.WithTx(setup.ls.DB, func(tx *leveldb.Transaction) error {
+					return store.SetHeaderTx(tx, &store.Header{
+						Name:        name,
+						EpochHeight: CurrentEpoch(name),
+						SectorSize:  10,
+						Banned:      true,
+						BannedAt:    time.Now().Add(-1 * 24 * time.Duration(time.Hour)),
+					}, blob.ZeroSectorHashes)
+				}))
+				err := UpdateBlob(cfg)
+				require.NotNil(t, err)
+				require.True(t, errors.Is(err, ErrInvalidEpochBackdated))
+			},
+		},
+		{
+			"aborts sync if the epoch is futuredated",
+			func(t *testing.T, setup *epochTestSetup) {
+				cfg := &UpdateConfig{
+					Mux:       setup.tp.LocalMux,
+					DB:        setup.ls.DB,
+					BlobStore: setup.ls.BlobStore,
+					Item: &UpdateQueueItem{
+						PeerIDs: NewPeerSet([]crypto.Hash{
+							crypto.HashPub(setup.tp.RemoteSigner.Pub()),
+						}),
+						Name:        name,
+						EpochHeight: CurrentEpoch(name) + 1,
+					},
+				}
+				err := UpdateBlob(cfg)
+				require.NotNil(t, err)
+				require.True(t, errors.Is(err, ErrInvalidEpochFuturedated))
+			},
+		},
+		{
+			"rewrites partial blob with new blob on epoch rollover",
+			func(t *testing.T, setup *epochTestSetup) {
+				ts := time.Now()
+				update := mockapp.FillBlobRandom(
+					t,
+					setup.rs.DB,
+					setup.rs.BlobStore,
+					setup.tp.RemoteSigner,
+					name,
+					CurrentEpoch(name),
+					blob.SectorCount,
+					ts,
+				)
+				cfg := &UpdateConfig{
+					Mux:        setup.tp.LocalMux,
+					DB:         setup.ls.DB,
+					NameLocker: util.NewMultiLocker(),
+					BlobStore:  setup.ls.BlobStore,
+					Item: &UpdateQueueItem{
+						PeerIDs: NewPeerSet([]crypto.Hash{
+							crypto.HashPub(setup.tp.RemoteSigner.Pub()),
+						}),
+						Name:          name,
+						EpochHeight:   update.EpochHeight,
+						SectorSize:    update.SectorSize,
+						SectorTipHash: update.SectorTipHash,
+						ReservedRoot:  update.ReservedRoot,
+						Signature:     update.Signature,
+						Pub:           setup.tp.RemoteSigner.Pub(),
+					},
+				}
+				require.NoError(t, store.WithTx(setup.ls.DB, func(tx *leveldb.Transaction) error {
+					return store.SetHeaderTx(tx, &store.Header{
+						Name:        name,
+						EpochHeight: CurrentEpoch(name) - 1,
+						SectorSize:  10,
+					}, blob.ZeroSectorHashes)
+				}))
+				require.NoError(t, UpdateBlob(cfg))
+				mockapp.RequireBlobsEqual(t, setup.ls.BlobStore, setup.rs.BlobStore, name)
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
Need simple tests for the `epoch.go` file above.

Reply:
Added simple test case in `epoch_test.go`